From a1690b446a77b7e8ed541af6ba603090fed38ed7 Mon Sep 17 00:00:00 2001
From: Known Rabbit <opensource@rabit.pw>
Date: Mon, 22 Dec 2025 11:26:21 +0800
Subject: [PATCH] fix(uuid): Improve UUID generation logic with modular
 arithmetic

Updated UUID generation to use modular arithmetic for version and variant bits.
---
 lua/uuid.lua | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lua/uuid.lua b/lua/uuid.lua
index 6252a515f..7659323c4 100644
--- a/lua/uuid.lua
+++ b/lua/uuid.lua
@@ -17,9 +17,9 @@ local function generate_uuid_v4()
 		rand(0, 255),
 		rand(0, 255),
 		rand(0, 255),
-		(rand(0, 255) & 0x0F) | 0x40,
+		((rand(0, 255) % 16) + 64),
 		rand(0, 255),
-		(rand(0, 255) & 0x3F) | 0x80,
+		((rand(0, 255) % 64) + 128),
 		rand(0, 255),
 		rand(0, 255),
 		rand(0, 255),
